go.mod

```diff
@@ -32,7 +32,7 @@ require (
 	github.com/onsi/gomega v1.10.2
 	github.com/opencontainers/image-spec v1.0.2-0.20190823105129-775207bd45b6
 	github.com/opencontainers/runc v0.1.1 // indirect
-	github.com/operator-framework/api v0.5.0
+	github.com/operator-framework/api v0.7.1
```

Comment:
Can we move this API bump to a separate PR? I don't see anything in this PR that requires the v0.7.x tag, but it's possible I'm missing something.

Reply:
The api bump is needed because the original purpose of the PR was to bring in validator changes, and then morphed from there as I did testing.

Reply:
Ah, gotcha that makes more sense. I think I would prefer if we still moved that bump to another PR, as we need to do it regardless of the changes in this PR, and have this PR depend on that PR, but this isn't a blocking concern for me. @joelanford Any strong opinion on whether we should land the API bump first, rather than throw it into this PR along with the other changes?

Reply:
Yeah I agree it would be good to separate the concerns of the PRs.

@jchunkins is there any coupling between the test changes and the API changes? If the test changes help validate the API changes, can the test changes go in first without bumping API?

Somewhat related, though not directly: I'm not sure how operator-registry has managed kubernetes version bumps in the past with respect to releases and branching, but IMO once a Kubernetes version bump lands, we _must_ not cut a patch release on an existing minor release. Put another way, every new Kubernetes version should automatically get at least a new minor release. I don't see any operator-registry-based version branching in the repo, so I'm concerned we're not currently doing this.

Reply:
Summary of the `API bump and its associated code`:

- I added a new test case "condition" to an existing test which exercises the validator changes coming from API v0.7.1. See this section https://github.com/operator-framework/operator-registry/pull/605/files#diff-fab19c28689357747c4449f5f7d8b9bd400867d1d096c65d656b07d6a5b4001fR166-R176

- Also, these items were added to make the test case work:

   pkg/lib/bundle/testdata/validate/invalid_manifests_bundle/invalid_annotation_name/etcdbackups.etcd.database.coreos.com.crd.yaml
   pkg/lib/bundle/testdata/validate/invalid_manifests_bundle/invalid_annotation_name/etcdclusters.etcd.database.coreos.com.crd.yaml
   pkg/lib/bundle/testdata/validate/invalid_manifests_bundle/invalid_annotation_name/etcdoperator.v0.9.4.clusterserviceversion.yaml
   pkg/lib/bundle/testdata/validate/invalid_manifests_bundle/invalid_annotation_name/etcdrestores.etcd.database.coreos.com.crd.yaml

`Everything else` is related to enhancements for running tests (i.e. TLS handling, kind & other server provisioning, etc.)

So I think you're asking me to strip out `API bump and associated code` into its own PR and get that merged first, then `Everything else` in this PR.

Reply:
Sorry for the slow reply. We just need to bump the API version listed in the root go.mod and re-generate the vendor tree (via `make vendor` locally) in a separate PR. If you want, I can throw up a PR that does that, as we realize we need that bump sooner rather than later before the openshift CI gets out of control in a couple of days. Else feel free to open that PR and I can that pushed through with the requisite labels.

Reply:
Thanks Tim. I created PR: https://github.com/operator-framework/operator-registry/pull/624

Reply:
Nice, that merged so we'll need to rebase this PR. Thanks for throwing that up 👍